File: app/backend/app/schemas/land_price.py
from pydantic import BaseModel


class LandPriceBase(BaseModel):
    prefecture_code: str
    prefecture_name: str
    city_code: str
    city_name: str
    district_name: str | None = None
    land_use: str | None = None
    price_per_sqm: int
    area_sqm: float | None = None
    year: int
    address: str | None = None
    station_name: str | None = None
    distance_to_station: int | None = None
    latitude: float | None = None
    longitude: float | None = None


class LandPriceCreate(LandPriceBase):
    pass


class LandPriceRead(LandPriceBase):
    id: int

    model_config = {"from_attributes": True}


class CityAnnualSummary(BaseModel):
    city_code: str
    city_name: str
    prefecture_name: str
    year: int
    avg_price_per_sqm: float
    min_price_per_sqm: int
    max_price_per_sqm: int
    record_count: int


class DistrictAnnualSummary(BaseModel):
    city_code: str
    city_name: str
    district_name: str
    year: int
    avg_price_per_sqm: float
    min_price_per_sqm: int
    max_price_per_sqm: int
    record_count: int

File: app/frontend/src/types/landPrice.ts
export interface LandPrice {
  id: number;
  prefecture_code: string;
  prefecture_name: string;
  city_code: string;
  city_name: string;
  district_name: string | null;
  land_use: string | null;
  price_per_sqm: number;
  area_sqm: number | null;
  year: number;
  address: string | null;
  station_name: string | null;
  distance_to_station: number | null;
  latitude: number | null;
  longitude: number | null;
}

export interface CityAnnualSummary {
  city_code: string;
  city_name: string;
  prefecture_name: string;
  year: number;
  avg_price_per_sqm: number;
  min_price_per_sqm: number;
  max_price_per_sqm: number;
  record_count: number;
}

export interface CityChangeRate {
  city_code: string;
  city_name: string;
  prefecture_name: string;
  baseYear: number;
  compareYear: number;
  avgPricePerSqm: number;
  changeRate: number | null;
}

export interface DistrictAnnualSummary {
  city_code: string;
  city_name: string;
  district_name: string;
  year: number;
  avg_price_per_sqm: number;
  min_price_per_sqm: number;
  max_price_per_sqm: number;
  record_count: number;
}

export interface DistrictChangeRate {
  city_code: string;
  city_name: string;
  district_name: string;
  baseYear: number;
  compareYear: number;
  avgPricePerSqm: number;
  changeRate: number | null;
}
